sw/device/tests/dif/dif_plic_sanitytest.c

```diff
@@ -117,131 +102,98 @@ static void uart_initialise(mmio_region_t base_addr, dif_uart_t *uart) {
+        "RX overflow trigger type set failed!");
+  CHECK(dif_plic_irq_trigger_type_set(plic, kTopEarlgreyPlicIrqIdUartTxEmpty,
+                                      kDifPlicDisable) == kDifPlicOk,
+        "TX empty trigger type set failed!");
 
   // Set IRQ priorities to MAX
-  if (dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdUartRxOverflow,
-                                kDifPlicMaxPriority) != kDifPlicOk) {
-    LOG_ERROR("priority set for RX overflow failed!");
-    return false;
-  }
+  CHECK(dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdUartRxOverflow,
+                                  kDifPlicMaxPriority) == kDifPlicOk,
+        "priority set for RX overflow failed!");
 
-  if (dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdUartTxEmpty,
-                                kDifPlicMaxPriority) != kDifPlicOk) {
-    LOG_ERROR("priority set for TX empty failed!");
-    return false;
-  }
+  CHECK(dif_plic_irq_priority_set(plic, kTopEarlgreyPlicIrqIdUartTxEmpty,
+                                  kDifPlicMaxPriority) == kDifPlicOk,
+        "priority set for TX empty failed!");
 
   // Set Ibex IRQ priority threshold level
-  if (dif_plic_target_threshold_set(&plic0, PLIC_TARGET, kDifPlicMinPriority) !=
-      kDifPlicOk) {
-    LOG_ERROR("threshold set failed!");
-    return false;
-  }
+  CHECK(dif_plic_target_threshold_set(&plic0, kPlicTarget,
+                                      kDifPlicMinPriority) == kDifPlicOk,
+        "threshold set failed!");
 
   // Enable IRQs in PLIC
-  if (dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdUartRxOverflow,
-                              PLIC_TARGET, kDifPlicEnable) != kDifPlicOk) {
-    LOG_ERROR("interrupt Enable for RX overflow failed!");
-    return false;
-  }
-  if (dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdUartTxEmpty,
-                              PLIC_TARGET, kDifPlicEnable) != kDifPlicOk) {
-    LOG_ERROR("interrupt Enable for TX empty failed!");
-    return false;
-  }
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdUartRxOverflow,
+                                kPlicTarget, kDifPlicEnable) == kDifPlicOk,
+        "interrupt Enable for RX overflow failed!");
 
-  return true;
+  CHECK(dif_plic_irq_enable_set(plic, kTopEarlgreyPlicIrqIdUartTxEmpty,
+                                kPlicTarget, kDifPlicEnable) == kDifPlicOk,
+        "interrupt Enable for TX empty failed!");
 }
 
-static bool execute_test(dif_uart_t *uart) {
-  // Initialize the global variables.
-  uart_rx_overflow_handled = false;
-  uart_tx_empty_handled = false;
-  uart_handled_more_than_once = false;
-
+static void execute_test(dif_uart_t *uart) {
   // Force UART RX overflow interrupt.
-  if (dif_uart_irq_force(uart, kDifUartInterruptRxOverflow) != kDifUartOk) {
-    LOG_ERROR("failed to force RX overflow IRQ!");
-    return false;
-  }
+  uart_rx_overflow_handled = false;
+  CHECK(dif_uart_irq_force(uart, kDifUartInterruptRxOverflow) == kDifUartOk,
```

Comment:
Yes. The test will still fail, though, but it will be slightly mysterious as to why. I think that we should keep this like it is for now and try to migrate the plic test to some other IP that won't get tripped by debug messages.

On the bright side, the CHECKs inside the interrupt handler can't trip the interrupt handler, so they'll run to completion and then abort the test.

Reply:
> I think that we should keep this like it is for now

You mean merge functionality in this PR, or not merge this PR? This statement is slightly ambiguous.

Reply:
Yeah, I meant to merge this PR for now, since the test ostensibly still works correctly, but we should make a note to fix it up to not use UART as the source of IRQs.

Reply:
> After the interrupts have been configured, won't any debug message trip an interrupts, and instead of this message you would get something from within the interrupt handler?

Could you please rephrase this and leave as a comment. Either in the interrupt handler description or in the test where it is forcing interrupts?

I think another thing to do, would be instead of `tx_empty` to use `rx_timeout` as it will never trip, unless we set it up. That way `rx_overflow` and `rx_timeout` are unlikely to fire unless forced. You don't have to change the IRQ to `rx_timeout`, we could do that as a follow-up.